### ISSUE TITLE
I want to compile a krokiet that uses translations from other languages. How should I do it?

### ISSUE THREAD

**Issue by chensimmons** [qarmin/czkawka#1284]
**Feature Description**

My English is very poor, I want to compile a krokiet with a non-English interface, what would I probably have to do?

**Comment by Uzume** [comment]
Can't you use the old method based upon a global callback and [projectfluent/fluent-rs](https://github.com/projectfluent/fluent-rs) as mentioned/alluded to  in that thread at: https://github.com/slint-ui/slint/issues/3715#issuecomment-1798221602?

**Comment by Uzume** [comment]
@qarmin: Can't you use the old method based upon a global callback and [projectfluent/fluent-rs](https://github.com/projectfluent/fluent-rs) as mentioned/alluded to in that issue thread at: https://github.com/slint-ui/slint/issues/3715#issuecomment-1798221602?

If you do not like that, [bundled translations](https://docs.slint.dev/latest/docs/slint/guide/development/translations/#bundled-translations) without `gettext` are still officially supported.

So you should be able to use translations without `gettext`, the question is whether you want to use the new `slint` translation syntax (i.e., `@tr` macro) and their officially supported bundled translations or use a slightly more cumbersome syntax and still leverage `fluent` and the existing `czkawka` translation architecture/content in a manner akin to #469 (especially if you name the global callback `fl` although `tr` seems nice too; many even `tr.fl` leaving room for other potential translation systems).

**Comment by qarmin** [comment]
I used a slightly different approach.
I implemented a Translations singleton that gets updated each time the language changes. All properties within it are linked to translatable elements in the GUI and this will be available in next version

**Comment by Uzume** [comment]
Yes, I see such support added with #1508 and the actual translations moved to that interface with #1513. Thank you.
